pkg/util/utils.go

```diff
@@ -396,6 +396,35 @@ func GetKeepIDMapping() (*stypes.IDMappingOptions, int, int, error) {
 	return &options, uid, gid, nil
 }
 
+// GetNoMapMapping returns the mappings and the user to use when nomap is used
+func GetNoMapMapping() (*stypes.IDMappingOptions, int, int, error) {
+	options := stypes.IDMappingOptions{
+		HostUIDMapping: true,
+		HostGIDMapping: true,
+	}
+	if rootless.IsRootless() {
+		uids, gids, err := rootless.GetConfiguredMappings()
+		if err != nil {
+			return nil, -1, -1, errors.Wrapf(err, "cannot read mappings")
+		}
+		options.UIDMap, options.GIDMap = nil, nil
+		uid, gid := 0, 0
+		for _, u := range uids {
+			options.UIDMap = append(options.UIDMap, idtools.IDMap{ContainerID: uid, HostID: uid + 1, Size: u.Size})
+			uid += u.Size
+		}
+		for _, g := range gids {
+			options.GIDMap = append(options.GIDMap, idtools.IDMap{ContainerID: gid, HostID: gid + 1, Size: g.Size})
+			gid += g.Size
+		}
+
+		options.HostUIDMapping = false
+		options.HostGIDMapping = false
+	}
+	// Simply ignore the setting and do not setup an inner namespace for root as it is a no-op
```

Comment:
Why should this be a noop for root? IMO it would make sense to use this as root when you want to make sure host uid 0 is not mapped into the container.
If this does not make sense for some reason it should error instead.

Reply:
For root you should use --userns=auto, which does the same thing, without allocating the entire namespace of UIDs.

Reply:
If you think so it should error, otherwise we are just ignoring user input.

Reply:
We are also currently ignoring the --userns=keep-id.